x/data/iri.go

```diff
@@ -111,3 +106,133 @@ var mediaTypeExtensions = map[MediaType]string{
 	MediaType_MEDIA_TYPE_WEBM:        "webm",
 	MediaType_MEDIA_TYPE_OGG:         "ogg",
 }
+
+var mediaTypeExtensionsReverse = map[string]MediaType{}
+
+func init() {
+	for mt, ext := range mediaTypeExtensions {
+		mediaTypeExtensionsReverse[ext] = mt
+	}
+}
+
+// ParseIRI parses an IRI string representation of a ContentHash into a ContentHash struct
+// Currently IRIs must have a "regen:" prefix, and only ContentHash_Graph and ContentHash_Raw
+// are supported.
+func ParseIRI(iri string) (*ContentHash, error) {
+	const regenPrefix = "regen:"
+
+	if !strings.HasPrefix(iri, regenPrefix) {
+		return nil, sdkerrors.ErrInvalidRequest.Wrap(fmt.Sprintf("can't parse IRI %s without %s prefix", iri, regenPrefix))
```

Comment:
@technicallyty we can simplify even further and just use wrapf. 

It would also be a good idea to add data module errors like InvalidIRI